src/rgw/rgw_rest_log.cc

```diff
@@ -396,19 +417,31 @@ void RGWOp_BILog_List::execute(optional_yield y) {
     }
   }
 
-  bool truncated;
+  // get next generation and num_shards
+  auto latest_gen = bucket_info.layout.logs.back().gen;
+  auto num_shards = bucket_info.layout.logs.back().layout.in_index.layout.num_shards;
```

Comment:
@cbodley ok. so currently in the code, `next_gen` is a` next_bilog_result` that contains a `generation` and `num_shards`. I could get those two values out of `std::optional<bucket_log_layout_generation>` `next_log` (probably change the name to next_log_layout) and pass them as part of send_response()

```C++
  const auto& logs = bucket_info.layout.logs;
  auto log = std::prev(logs.end());
  if (gen) {
    log = std::find_if(logs.begin(), logs.end(), rgw::matches_gen(gen));
    if (log == logs.end()) {
      ldpp_dout(s, 5) << "ERROR: no log layout with gen=" << gen << dendl;
      op_ret = -ENOENT;
      return;
    }
  }
  if (auto next = std::next(log); next != logs.end()) {
    next_log_layout = *next;    // get the next log after the current latest
  }
  auto log_layout = *log; // current log layout for log listing
```
I get the gen and num_shards from `next_log_layout` and make that part of send_response: 
```C++
    s->formatter->open_object_section("next_log");
    encode_json("generation",latest_gen, s->formatter);
    encode_json("num_shards", num_shards, s->formatter);
    s->formatter->close_section(); // next_log
```
But are you suggesting instead of this, we just send the std::optional<bucket_log_layout_generation> as response?

Reply:
the json format would still look the same with "generation" and "num_shards", the only difference would be that we're getting those values from a `bucket_log_layout_generation`

Reply:
updated, thanks!